indexing-service/src/main/java/io/druid/indexing/common/task/SinglePhaseParallelIndexSupervisorTask.java

```diff
@@ -0,0 +1,489 @@
+
+  private volatile boolean stopped;
+  private int nextSpecId = 0;
+
+  @JsonCreator
+  public SinglePhaseParallelIndexSupervisorTask(
+      @JsonProperty("id") String id,
+      @JsonProperty("resource") TaskResource taskResource,
+      @JsonProperty("spec") SinglePhaseParallelIndexIngestionSpec ingestionSchema,
+      @JsonProperty("context") Map<String, Object> context,
+      @JacksonInject @Nullable IndexingServiceClient indexingServiceClient, // null in overlords
+      @JacksonInject @Nullable ChatHandlerProvider chatHandlerProvider,     // null in overlords
+      @JacksonInject AuthorizerMapper authorizerMapper
+  )
+  {
+    super(
+        getOrMakeId(id, TYPE, ingestionSchema.getDataSchema().getDataSource()),
+        null,
+        taskResource,
+        ingestionSchema.getDataSchema().getDataSource(),
+        context
+    );
+
+    this.ingestionSchema = ingestionSchema;
+
+    final FirehoseFactory firehoseFactory = ingestionSchema.getIOConfig().getFirehoseFactory();
+    if (!(firehoseFactory instanceof FiniteFirehoseFactory)) {
+      throw new IAE("[%s] should implement FiniteFirehoseFactory", firehoseFactory.getClass().getSimpleName());
+    }
+
+    this.baseFirehoseFactory = (FiniteFirehoseFactory) firehoseFactory;
+    this.maxNumTasks = ingestionSchema.getTuningConfig().getMaxNumBatchTasks();
+    this.indexingServiceClient = indexingServiceClient;
+    this.chatHandlerProvider = chatHandlerProvider;
+    this.authorizerMapper = authorizerMapper;
+  }
+
+  @Override
+  public int getPriority()
+  {
+    return getContextValue(Tasks.PRIORITY_KEY, Tasks.DEFAULT_BATCH_INDEX_TASK_PRIORITY);
+  }
+
+  @Override
+  public String getType()
+  {
+    return TYPE;
+  }
+
+  @JsonProperty("spec")
+  public SinglePhaseParallelIndexIngestionSpec getIngestionSchema()
+  {
+    return ingestionSchema;
+  }
+
+  @Override
+  public boolean isReady(TaskActionClient taskActionClient) throws Exception
+  {
+    final Optional<SortedSet<Interval>> intervals = ingestionSchema.getDataSchema()
+                                                                   .getGranularitySpec()
+                                                                   .bucketIntervals();
+
+    return !intervals.isPresent() || isReady(taskActionClient, intervals.get());
+  }
+
+  static boolean isReady(TaskActionClient actionClient, SortedSet<Interval> intervals) throws IOException
+  {
+    final List<TaskLock> locks = getTaskLocks(actionClient);
+    if (locks.size() == 0) {
+      try {
+        Tasks.tryAcquireExclusiveLocks(actionClient, intervals);
+      }
+      catch (Exception e) {
+        log.error(e, "Failed to acquire locks for intervals[%s]", intervals);
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public TaskStatus run(TaskToolbox toolbox) throws Exception
+  {
+    if (baseFirehoseFactory.isSplittable()) {
+      return runParallel(toolbox);
+    } else {
+      log.warn(
```

Comment:
I thought this task type will eventually replace with the existing local index task in the future. I think as-is would be better because users don't have to worry about which task type is required for each firehose type.